libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -98,14 +101,29 @@ export class StripeManager {
   }
 
   async retrieveSubscription(subscriptionId: string) {
-    return this.client.subscriptionsRetrieve(subscriptionId);
+    try {
+      const subscription = await this.client.subscriptionsRetrieve(
+        subscriptionId
+      );
+      return subscription;
+    } catch (err) {
+      throw new SubscriptionNotFoundError();
```

Comment:
These should retain the original error so if we did want to report these to sentry we have the full stack trace with the original error.